backend/utils/cache_service.py

```diff
@@ -28,6 +28,12 @@ def set_key(
             expire,
         )
 
+    @staticmethod
+    def get_all_keys(key_pattern: str) -> Any:
+        keys = redis_cache.keys(key_pattern)
+        # Ensure all keys are strings
+        return [key.decode("utf-8") if isinstance(key, bytes) else key for key in keys]
+
```

Comment:
I think, It block redis conenctions if we have large number of keys. We can implement following later to optimize this
```
while cursor != 0:
   redis_client.scan(cursor=cursor, match=key_pattern, count=<count>)
```

Reply:
Yes, will be done on v2